modules/mono/glue/GodotSharp/GodotSharp/Core/StringExtensions.cs

```diff
@@ -428,6 +431,7 @@ public static bool EndsWith(this string instance, string text)
         /// <param name="instance">The string to modify.</param>
         /// <param name="pos">Starting position from which to erase.</param>
         /// <param name="chars">Amount of characters to erase.</param>
+        [Obsolete("Use StringBuilder.Remove instead.")]
```

Comment:
This method was always iffy because strings are immutable. I guess I added it for StringBuilder instead of wrapper `string.Remove` because I was afraid that users would forget about the returned string. Since we're deprecating it, we may as well recommend `string.Remove` and point out that a new string is returned.

Reply:
I'm not sure if there's a lot of people using this method (specially since it's on `StringBuilder`) and, to be honest, in 4.0 it was completely removed from GDScript. Since this is still 3.x I didn't want to add an annoying warning without an actionable step for users to take to get rid of it.

I can add `string.Remove` to the recommended alternatives, but if any user was using this extension method then they'll already be using `StringBuilder` anyway.

I can also point out that a new string is returned but the documentation for `string.Remove` already mentions this.